README.md

```diff
@@ -748,6 +752,54 @@ The documentation on using Closure Stylesheets can be found
   `bazel run @io_bazel_rules_closure//closure/stylesheets -- --help`
 
 
+## closure\_proto\_js\_library
+
+```python
+load("@io_bazel_rules_closure//closure:defs.bzl", "closure_proto_js_library")
+closure_proto_js_library(name, srcs, add_require_for_enums, namespace_prefix,
+                         error_on_name_conflict, binary, import_style)
+```
+
+Defines a set of Protocol Buffer files.
+
+Documentation: [Protocol Buffers][protobuf] [JS][protobuf-js]
+[Generator Options](https://github.com/google/protobuf/blob/master/src/google/protobuf/compiler/js/js_generator.h).
+
+#### Implicit Output Targets
+
+- *name*.js: A generated protocol buffer JavaScript library.
+
+### Arguments
+
+- **name:** ([Name][name]; required) A unique name for this rule. Convention
+  states that such rules be named `foo_proto`.
+
+- **srcs:** (List of [labels][labels]; required) A list of `.proto` source
+  files that represent this library.
+
+- **add_require_for_enums:** (Boolean; optional; default is `False`) Add a
```

Comment:
Unless I mis-understand (which is very possible) when a type is forward declared (but not required elsewhere) the compiler's level of type checking is binary (either the type exists or it does not). The compiler knows nothing else about the implementation of the type (eg: data members, functions, etc). However, as the compiler works on all sources at a time I think the most common (only?) case this can happen in is code that takes a type as a function parameter but does not need to require it that type. I think this happens most frequently (only?) for code which is not called. If the code were called some other file would have needed to instantiate said type -- hence creating an explicit dependency and enabling the compilers full type checking.

Do you know of more complicated cases?

I agree that a lot more thought and discussion is needed here and an issue tracker item would be good.